SimG4Core/CustomPhysics/src/FullModelHadronicProcess.cc

```diff
@@ -1,45 +1,42 @@
-#include "G4ProcessManager.hh"
-#include "G4ParticleTable.hh"
-#include "G4HadronicException.hh"
-
 #include "SimG4Core/CustomPhysics/interface/FullModelHadronicProcess.h"
-#include "SimG4Core/CustomPhysics/interface/G4ProcessHelper.h"
+#include "SimG4Core/CustomPhysics/interface/CustomProcessHelper.h"
 #include "SimG4Core/CustomPhysics/interface/Decay3Body.h"
 #include "SimG4Core/CustomPhysics/interface/CustomPDGParser.h"
 #include "SimG4Core/CustomPhysics/interface/CustomParticle.h"
 
-using namespace CLHEP;
+#include "G4ProcessManager.hh"
+#include "G4ParticleTable.hh"
+#include "G4Material.hh"
+#include "G4Element.hh"
+#include "G4Isotope.hh"
 
-FullModelHadronicProcess::FullModelHadronicProcess(G4ProcessHelper* aHelper, const G4String& processName)
-    : G4VDiscreteProcess(processName), theHelper(aHelper) {}
+using namespace CLHEP;
 
-FullModelHadronicProcess::~FullModelHadronicProcess() {}
+FullModelHadronicProcess::FullModelHadronicProcess(CustomProcessHelper* aHelper, const G4String& pname)
+    : G4VDiscreteProcess(pname), theHelper(aHelper) {
+  xsec_.reserve(12);
+}
 
 G4bool FullModelHadronicProcess::IsApplicable(const G4ParticleDefinition& aP) {
   return theHelper->ApplicabilityTester(aP);
 }
 
-G4double FullModelHadronicProcess::GetMicroscopicCrossSection(const G4DynamicParticle* aParticle,
-                                                              const G4Element* anElement,
-                                                              G4double aTemp) {
-  //Get the cross section for this particle/element combination from the ProcessHelper
-  G4double InclXsec = theHelper->GetInclusiveCrossSection(aParticle, anElement);
-  return InclXsec;
-}
-
-G4double FullModelHadronicProcess::GetMeanFreePath(const G4Track& aTrack, G4double, G4ForceCondition*) {
+G4double FullModelHadronicProcess::GetMeanFreePath(const G4Track& aTrack, G4double, G4ForceCondition* cond) {
+  *cond = NotForced;
   G4Material* aMaterial = aTrack.GetMaterial();
   const G4DynamicParticle* aParticle = aTrack.GetDynamicParticle();
   G4double sigma = 0.0;
 
-  G4int nElements = aMaterial->GetNumberOfElements();
+  std::size_t nElements = aMaterial->GetNumberOfElements();
+  xsec_.resize(nElements);
 
   const G4double* theAtomicNumDensityVector = aMaterial->GetAtomicNumDensityVector();
-  G4double aTemp = aMaterial->GetTemperature();
 
-  for (G4int i = 0; i < nElements; ++i) {
-    G4double xSection = GetMicroscopicCrossSection(aParticle, (*aMaterial->GetElementVector())[i], aTemp);
+  for (std::size_t i = 0; i < nElements; ++i) {
+    auto anElement = (*aMaterial->GetElementVector())[i];
+    G4double xSection = theHelper->GetInclusiveCrossSection(aParticle, anElement);
```

Comment:
@civanch can you explain this change, which seems like it could potentially impact the physics?